src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLTripImpl.java

```diff
@@ -214,14 +215,23 @@
       try {
         RoutingService routingService = getRoutingService(environment);
         Trip trip = getSource(environment);
-        TripPattern tripPattern = getTripPattern(environment);
-        if (tripPattern == null) { return List.of(); }
-
         var args = new LegacyGraphQLTypes.LegacyGraphQLTripStoptimesForDateArgs(environment.getArguments());
 
-
         ServiceDate serviceDate = args.getLegacyGraphQLServiceDate() != null
-            ? ServiceDate.parseString(args.getLegacyGraphQLServiceDate()) : new ServiceDate();
+                ? ServiceDate.parseString(args.getLegacyGraphQLServiceDate()) : new ServiceDate();
+
+        TripPattern tripPattern;
+        TimetableSnapshot timetableSnapshot = routingService.getTimetableSnapshot();
```

Comment:
Since your username is @optionsome could you not use an `Optional` chain here to make this easier to read?

Reply:
Not, no not ;-), sure if I understand what you mean here @leonardehrenfried? Is this related to the code above or below?

Reply:
Below. I'm not a huge fan of repeated null checks. A chain of Optionals reads much better, IMO.

Reply:
So this what you want:
```java
tripPattern = Optional
      .ofNullable(routingService.getTimetableSnapshot())
      .map(timetableSnapshot -> timetableSnapshot.getLastAddedTripPattern(trip.getId(), serviceDate))
      .orElseGet(() -> getTripPattern(environment));
```
Even if it is much shorter I think it is within the "license of the author" to decide here, note that the code could also be a bit shorter:
```java
if (timetableSnapshot != null) {
  tripPattern = timetableSnapshot.getLastAddedTripPattern(trip.getId(), serviceDate);
}
// timetableSnapshot is null or pattern not found
if (tripPattern == null) {
  tripPattern = getTripPattern(environment);
}
```
I think I prefer the second version here, but, I will leave this on tho the "author"

Reply:
I prefer the second alt. because:
- it is not obvious what the map method does - it take a second to understand
- there are easy room for mistakes like this:
```
tripPattern = Optional
      .ofNullable(routingService.getTimetableSnapshot())
      .map(timetableSnapshot -> timetableSnapshot.getLastAddedTripPattern(trip.getId(), serviceDate))
      .orElse(getTripPattern(environment));
```

Reply:
I tend to agree with @t2gran here, I find the chain of optionals to be more difficult to read. I did the minor change suggested by Thomas https://github.com/opentripplanner/OpenTripPlanner/pull/3960/commits/7852b8ba48f4be7e750247bc230cd31b331d3245